mq/plugins/lower_keys.py

```diff
@@ -0,0 +1,31 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# Copyright (c) 2014 Mozilla Corporation
+
+
+class message(object):
+
+    def __init__(self):
+        '''
+        takes an incoming message
+        and sets the keys to lowercase
+        '''
+
+        self.registration = ['summary']
+        self.priority = 5
+
+    def onMessage(self, message, metadata):
+        def renameKeysToLower(message):
+            if isinstance(message, dict):
```

Comment:
I think this isn't going to traverse into really complex sub structures like are present in cloudtrail? (lists of dicts of lists, etc)
https://github.com/mozilla/MozDef/blob/master/mozdef_util/mozdef_util/utilities/dict2List.py is an example of some of the complexity involved in this.

Reply:
It does, I have tested it on actual cloudtrail events.  Thanks for the inquiry!